core/Middleware/MiddlewareMap.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\generis\model\Middleware;
+
+final class MiddlewareMap implements MiddlewareMapInterface
+{
+    private const OPTION_MIDDLEWARES = 'middlewares';
+    private const OPTION_ROUTES = 'routes';
+    private const OPTION_HTTP_METHODS = 'httpMethods';
+
+    /** @var string[] */
+    private $routes = [];
+
+    /** @var string[] */
+    private $middlewaresIds = [];
+
+    /** @var string[]|null */
+    private $httpMethods = [];
+
+    public static function byRoute(string $route): self
+    {
+        return new self([$route]);
+    }
+
+    public static function byRoutes(string ...$route): self
+    {
+        return new self($route);
+    }
+
+    public static function byMiddlewareId(string $middlewareId): self
+    {
+        return new self([], [$middlewareId]);
+    }
+
+    public static function byMiddlewareIds(string ...$middlewareIds): self
+    {
+        return new self([], $middlewareIds);
+    }
```

Comment:
Maybe it's better to use factory (or builder) instead?
And then we can make `__construct` public

Wdyt?

Reply:
Not necessary for this case IMO. Builder is a patter and in this case the static method is a builder